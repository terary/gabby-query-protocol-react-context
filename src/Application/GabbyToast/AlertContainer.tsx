import Alert from "@mui/material/Alert";
import AlertTitle from "@mui/material/AlertTitle";

type AlertContainerProps = {
  message: string;
  title?: string;
  severity: "error" | "info" | "success" | "warning";
};

const AlertContainer = ({ message, title = "Info", severity }: AlertContainerProps) => {
  return (
    <Alert severity={severity} sx={{ backgroundColor: "rgba(255,255,255,0.5)" }}>
      <AlertTitle>{title}</AlertTitle>
      {message} â€” <strong>check it out!</strong>
    </Alert>
  );
};

export { AlertContainer };
