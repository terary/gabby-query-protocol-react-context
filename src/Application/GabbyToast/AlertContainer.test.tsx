import React from "react";
import renderer from "react-test-renderer";
import { AlertContainer } from "./AlertContainer";
import { BrowserRouter } from "react-router-dom";

describe("AlertContainer (internal use only)", () => {
  it("Should render error", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AlertContainer
            message="Alert Container Message"
            title="The Title"
            severity="error"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should render info", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AlertContainer
            message="Alert Container Message"
            title="The Title"
            severity="info"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should render success", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AlertContainer
            message="Alert Container Message"
            title="The Title"
            severity="success"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should render warning", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AlertContainer
            message="Alert Container Message"
            title="The Title"
            severity="warning"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Title should be optional", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AlertContainer message="Alert Container Message" severity="warning" />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
