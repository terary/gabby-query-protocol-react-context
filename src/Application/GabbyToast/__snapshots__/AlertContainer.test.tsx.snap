// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertContainer (internal use only) Should render error 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1ydba47-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ErrorOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-acap47-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      The Title
    </div>
    Alert Container Message
     — 
    <strong>
      check it out!
    </strong>
  </div>
</div>
`;

exports[`AlertContainer (internal use only) Should render info 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1jy379c-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="InfoOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-acap47-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      The Title
    </div>
    Alert Container Message
     — 
    <strong>
      check it out!
    </strong>
  </div>
</div>
`;

exports[`AlertContainer (internal use only) Should render success 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-4jc1xm-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="SuccessOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-acap47-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      The Title
    </div>
    Alert Container Message
     — 
    <strong>
      check it out!
    </strong>
  </div>
</div>
`;

exports[`AlertContainer (internal use only) Should render warning 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-jamnmr-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-acap47-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      The Title
    </div>
    Alert Container Message
     — 
    <strong>
      check it out!
    </strong>
  </div>
</div>
`;

exports[`AlertContainer (internal use only) Title should be optional 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-jamnmr-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-acap47-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      Info
    </div>
    Alert Container Message
     — 
    <strong>
      check it out!
    </strong>
  </div>
</div>
`;
